assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -318,6 +318,56 @@ const ExternalActivity: React.FC<any> = (props) => {
     }
   };
 
+  const getInterestedVars = (vars: any) => {
+    const interested = Object.keys(vars).filter((ms: any) => {
+      const isMine = ms.indexOf(`stage.${id}.`) === 0;
+      if (!isMine) {
+        return false;
+      }
+      const internalValue = externalActivityMap.get(ms);
+      let mineValue = vars[ms];
+      let internalVal: any = internalValue?.value;
+
+      if (ms.type === CapiVariableTypes.BOOLEAN && typeof internalVal === 'string') {
+        mineValue = JSON.stringify(mineValue);
+      }
+      if (ms.type === CapiVariableTypes.NUMBER && typeof internalVal === 'string') {
```

Comment:
`typeof internalVal === 'string'` is used several times, good to factor into a variable at the top